### ISSUE TITLE
Quick fix for test instability around Test.IdPrefix

### ISSUE THREAD

**Issue by OmicronPersei** [nunit/nunit#2986]
As noted in #2963 , some tests in the solution are changing `Test.IdPrefix` to a value that's not expected by other tests, causing intermittent test failures, due to the fact that `IdPrefix` is a static field.  As a quick fix, we should go through all tests that set `IdPrefix` to a invalid value and revert to its original value on test teardown.

**Comment by OmicronPersei** [comment]
So far it's looking like simply adding a `TearDown` method that resets `Test.IdPrefix` isn't stopping the ocassional exceptions.

@nunit/framework-team , does anyone know if our build script (`build.ps1`) runs tests in parallel by default?

**Comment by ChrisMaddock** [comment]
I don’t think it does...from memory. Search the code base for parallelizable attributes?

**Comment by jnm2** [comment]
I'd like to be able to run our tests in parallel in VS though. Can you put `[NonParallelizable]` on the test methods that change `Test.IdPrefix`? That way we retain parallelism for most tests but don't have to worry about any tests running in parallel with those tests.

**Comment by CharliePoole** [comment]
@jnm2 I recommend we focus in fixing the root problem if we are worried about the future. The prefix should not be a static in the first place.

For parallelizing our own tests, I would do the reverse... mark those that can be parallelized. NUnit needs a lot of work before we use a default of parallel for all methods.

**Comment by OmicronPersei** [comment]
After thinking about it some more, we probably should devote effort towards fixing the underlying issue rather than creating a fix in this manner.

The fact that our tests are showing this error is arguably a good thing, by finding a potential deficiency in the design of our code.  So creating a quick fix might not be a good idea.

**Comment by jnm2** [comment]
@OmicronPersei Due to your investigation, I discovered that the SchemaTests which I added were at fault and decided to fix them. Kind of a quick fix after all. I still want to pursue the deeper fix @CharliePoole mentions but it's looking like a lot of talking will be needed: https://github.com/nunit/nunit/issues/2963#issuecomment-416946415

So this gets CI bearable and cleans up some my own sloppiness: https://github.com/nunit/nunit/pull/2993

**Comment by CharliePoole** [comment]
Here's a related somewhat basic question: why should any test change the test id prefix? Is it simply that you had to provide something as an argument to the constructor of `FrameworkController`? If so, maybe we should redefine null to mean "don't set it".

**Comment by jnm2** [comment]
FrameworkControllerTests depend on being able to reset Test.IdFilter to `"ID"`.
